### PR TITLE
Do refactoring and simplify the sync code

### PR DESCRIPTION
**Pull request by jhs7jhs**
Removed redundant code, and made the flow clear from the hotfix #712

### CODE REVIEW

File: sync/src/block/extension.rs [hunk 1]
```diff
@@ -46,12 +46,18 @@ const SYNC_EXPIRE_REQUEST_INTERVAL: i64 = 15000;
 
 const SNAPSHOT_PERIOD: u64 = (1 << 14);
 
+#[derive(Debug, PartialEq)]
+pub struct TokenInfo {
+    id: NodeId,
```
Comment (sgkim126) [inline sync/src/block/extension.rs]: How about `node_Id`?

File: sync/src/block/extension.rs [hunk 2]
```diff
@@ -267,28 +267,36 @@ impl NetworkExtension for Extension {
...
+                    (id, request_id)
+                };
 
-                let expired_request = request_list.iter().find(|(r, _)| *r == request_id).cloned();
-                if let Some((request_id, request)) = expired_request {
-                    match request {
-                        RequestMessage::Bodies(hashes) => {
-                            self.body_downloader.lock().reset_downloading(&hashes);
+                if let Some(requests) = self.requests.write().get_mut(&id) {
+                    let expired_request = requests.iter().find(|(r, _)| *r == request_id).cloned();
```
Comment (sgkim126) [inline sync/src/block/extension.rs]: It's not about this patch, but there is no reason to call `cloned()`.

File: sync/src/block/extension.rs (hunk 1 above)
Comment (jhs7jhs) [reply sync/src/block/extension.rs]: Good idea.

File: sync/src/block/extension.rs [hunk 3]
```diff
@@ -267,28 +267,36 @@ impl NetworkExtension for Extension {
                 }
             }
             SYNC_EXPIRE_TOKEN_BEGIN...SYNC_EXPIRE_TOKEN_END => {
-                let mut requests = self.requests.write();
-                let mut tokens_info = self.tokens_info.write();
-                let token_info = tokens_info.get_mut(&token).unwrap();
-                if token_info.request_id.is_none() {
-                    return
-                }
+                let (id, request_id) = {
```
Comment (sgkim126) [inline sync/src/block/extension.rs]: It seems that it's not just a refactoring, but changing the behavior.
Is it safe to not lock `requests` while checking `tokens_info`?

Comment (jhs7jhs) [reply sync/src/block/extension.rs]: It is not refactoring, it's my mistake. And, yes, it is safe.

File: sync/src/block/extension.rs (hunk 2 above)
Comment (jhs7jhs) [reply sync/src/block/extension.rs]: Fixed

File: sync/src/block/extension.rs (hunk 3 above)
Comment (jhs7jhs) [reply sync/src/block/extension.rs]: There was no reason to lock `tokens_info` after locking `requests`, and the key point of this patch is to make `token_info.request_id.is_none()` as a flag between `on_timeout()` and `on_peer_response()`.

Comment (sgkim126) [reply sync/src/block/extension.rs]: Okay.
@Kais-DkM how about you?

File: sync/src/block/extension.rs [hunk 4]
```diff
@@ -257,28 +267,36 @@ impl NetworkExtension for Extension {
...
+                    let mut tokens_info = self.tokens_info.write();
+                    let token_info = tokens_info.get_mut(&token);
+                    if token_info.is_none() {
+                        return
+                    }
 
-                let id = token_info.0;
-                let request_id = token_info.1.unwrap();
-                let request_list = requests.get_mut(&id).unwrap();
+                    let token_info = token_info.unwrap();
```
Comment (Kais-DkM) [inline sync/src/block/extension.rs]: How about this one?
```
let token_info = match tokens_info.get_mut(&token) {
    Some(info) => info,
    None => return,
};
```

File: sync/src/block/extension.rs (hunk 3 above)
Comment (Kais-DkM) [reply sync/src/block/extension.rs]: It seems to be safe at a first glance. I'll accept it.

File: sync/src/block/extension.rs (hunk 4 above)
Comment (jhs7jhs) [reply sync/src/block/extension.rs]: Much better! Leaning a lot +1

**Pull request merged**